app/src/main/java/org/groundplatform/android/ui/datacollection/DataCollectionViewModel.kt

```diff
@@ -209,6 +210,7 @@ internal constructor(
       val taskData = if (shouldLoadFromDraft) getValueFromDraft(state.job, task) else null
       created.initialize(state.job, task, taskData)
       taskDataHandler.setData(task, taskData)
+      taskSequenceHandler.invalidateCache()
```

Comment:
Can you move these lines to a new method and reuse that method across all instances in this class?

Reply:
done

Reply:
I meant moving both lines since the issue is originating from using setData without invalidating. 

```
fun updateDataAndInvalidateTasks(task, taskData) {
taskDataHandler.setData(task, taskData)
taskSequenceHandler.invalidateCache()
}
``